scenario_player/tasks/services.py

```diff
@@ -289,8 +289,9 @@ def _process_response(self, response_dict: dict):
         if self._config.get("iou_exists", True) is False:
             if response_dict:
                 raise ScenarioAssertionError(f"Expected no IOU but got {response_dict}.")
-
-        if "amount" in self._config and "amount" in response_dict:
+        elif not response_dict:
```

Comment:
the iou debug endpoint will always return a dict, see [here](https://github.com/raiden-network/raiden-services/blob/e254d894994f9db6d14e7fcc5fc5dbbf5d44af34/src/pathfinding_service/api.py#L390). Which means `response_dict` is always true, right?

Reply:
May use `elif "amount" in self._config and not "amount" in response_dict:` That way you cover the case where the PFS does not have an IOU and returns `{}` but the task expects an IOU

Reply:
> the iou debug endpoint will always return a dict, see [here](https://github.com/raiden-network/raiden-services/blob/e254d894994f9db6d14e7fcc5fc5dbbf5d44af34/src/pathfinding_service/api.py#L390). Which means `response_dict` is always true, right?

No, if you defint a dict `dct = {}` and then do `not dct` you get `True`, so this should be fine.